zbsprimarystorage/zbsprimarystorage/zbsagent.py

```
@@ -0,0 +1,528 @@
+                copyset_number=d.copysetNum,
+                replica_number=d.replicaNum,
+                zone_number=d.zoneNum
+            )
+        except Exception as e:
+            logger.error('failed to decode redundance and placement policy[%s], error[%s]' % (
+                redundance_and_placement_policy, e.message
+            ))
+
+
+def replyerror(func):
+    @functools.wraps(func)
```

Ruff (0.8.2)

`functools` may be undefined, or defined from star imports

(F405)